src/components/VMapbox.vue

```diff
@@ -211,10 +219,11 @@ export default {
       this.addLayers()
     })
 
-    // If the style was changed, wait for the styledata to be loaded and re-add all the layers
+    // If the style was changed, wait for the style to be loaded and re-add all the layers
+    // https://github.com/mapbox/mapbox-gl-js/issues/4006
     this.$on('style:update', () => {
-      this.$once('mb-styledata', () => {
-        this.addLayers()
+      this.$once('mb-style.load', () => {
+        this.refreshLayers()
```

Comment:
Only the map layers get removed when the style changes, `addLayers` duplicated the controllers